src/Single.php

```diff
@@ -366,6 +366,29 @@ public static function map(iterable $data, callable $func): \Generator
         }
     }
 
+    /**
+     * Returns a new collection formed by applying a given callback function to each element
+     * of the given collection, and then flattening the result by one level.
+     *
+     * @param iterable<mixed> $data
+     * @param callable $func
+     *
+     * @return \Generator
+     */
+    public static function flatMap(iterable $data, callable $func): \Generator
+    {
+        foreach ($data as $datum) {
+            $unflattened = $func($datum, $func);
+            if (is_iterable($unflattened)) {
+                foreach ($func($datum, $func) as $unflattenedItem) {
+                    yield $unflattenedItem;
```

Comment:
This foreach loop actually seems wrong, although all the tests pass, so maybe I'm thinking about it incorrectly.

I changed it to this in the `develop` branch:
```php
        foreach ($data as $datum) {
            $unflattened = $func($datum, $func);
            if (\is_iterable($unflattened)) {
                foreach ($unflattened as $flattenedItem) {
                    yield $flattenedItem;
                }
            } else {
                yield $unflattened;
            }
        }
```

Take a look at that and see if that makes sense. This seems to me to be the more intuitive way to write this. Let me know if this is a problem.

Reply:
Thanks @markrogoyski! At the last moment (after merging PR), I noticed that there was a repeated call to the function, I wanted to open a fix-PR, but I saw that you had already fixed this thing.